homeassistant/components/switch/freebox.py

```diff
@@ -0,0 +1,93 @@
+"""
+Support for Freebox devices (Freebox v6 and Freebox mini 4K).
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.freebox/
+"""
+import logging
+
+from homeassistant.components.freebox import DATA_FREEBOX
+from homeassistant.components.switch import SwitchDevice
+
+DEPENDENCIES = ['freebox']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensors."""
+    fbx = hass.data[DATA_FREEBOX]
+    async_add_entities([FbxWifiSwitch(fbx)], True)
+
+
+class FbxSwitch(SwitchDevice):
+    """Representation of a Freebox switch."""
+
+    _name = 'generic'
+
+    def __init__(self, fbx):
+        """Initilize the switch."""
+        self._state = None
+        self._fbx = fbx
+        self._permissions = None
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    async def _async_update_perms(self):
+        """Get permissions from Freebox."""
+        self._permissions = await self._fbx.get_permissions()
```

Comment:
If the permissions are the same for all entitites, we could check permissions in the component already and store them in hass.data.

Reply:
See my other answer I guess it answer this too.